android/src/main/java/com/pylon/RNSpokestack/RNSpokestackModule.java

```diff
@@ -17,101 +19,138 @@
 import java.util.Map;
 import javax.annotation.Nullable;
 
-public class RNSpokestackModule extends ReactContextBaseJavaModule implements OnSpeechEventListener {
+public class RNSpokestackModule extends ReactContextBaseJavaModule implements OnSpeechEventListener, TTSListener {
 
-  private final ReactApplicationContext reactContext;
-  private SpeechPipeline pipeline;
+    private final ReactApplicationContext reactContext;
+    private SpeechPipeline pipeline;
+    private TTSManager tts;
 
-  public RNSpokestackModule(ReactApplicationContext reactContext) {
-    super(reactContext);
-    this.reactContext = reactContext;
-  }
+    public RNSpokestackModule(ReactApplicationContext reactContext) {
+        super(reactContext);
+        this.reactContext = reactContext;
+    }
+
+    @Override
+    public String getName() {
+        return "RNSpokestack";
+    }
+
+    @Override
+    public void onCatalystInstanceDestroy() {
+        super.onCatalystInstanceDestroy();
+        if (pipeline != null && pipeline.isRunning()) {
+            pipeline.stop();
+        }
+    }
+
+    private void sendEvent(String eventName, @Nullable WritableMap params) {
+        if (this.reactContext.hasActiveCatalystInstance()) {
+            this.reactContext
+                    .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
+                    .emit(eventName, params);
+        }
+    }
 
-  @Override
-  public String getName() {
-    return "RNSpokestack";
-  }
+    @ReactMethod
+    public void initialize(ReadableMap config) throws Exception {
+
+        // SpeechPipeline
+        if (config.hasKey("input")) {
```

Comment:
I've thought about this, and concluded that since the pipeline, tts, and nlu components can operate independently, no error should be thrown from these `if` checks in `initialize`. Assume that the client knows what they're doing. As for null checks for the `pipeline`/`tts`/`nlu` managers in the methods, I don't think that's necessary either since they will result in a very obvious NPE if called when uninitialized (which I myself experienced while developing this interface. It was easy to debug what happened).